### PR TITLE
c-deps: support 16K pages for ARM64

### PR DESCRIPTION
**Pull request by RaduBerinde**
We have had two reports of users not being able to run cockroach 23.1 on linux/arm64 with 16k pages (jemalloc does not support the page size). This change attempts to fix this by compiling jemalloc with 16K page support, only in the linux/arm64 configuration.

Epic: None
Informs: #106745
Release note: None

### CODE REVIEW

**Review by rickystewart — APPROVED** [review]
This is fine but we need to re-build the pre-built `c-deps`. I'll wait until #106642 is merged to do that because we'll get conflicts otherwise. This can merge in the meantime (or you can wait until #106642 is merged, if you want to get everything in in one big commit).

Either way can you make sure to backport to 23.1?

**Comment by RaduBerinde** [comment]
Thanks @rickystewart! Yes, I plan to backport to 23.1.

bors r+

**Comment by RaduBerinde** [comment]
blathers backport 23.1

**Pull request merged**
